internal/dcache/rpc/server/handler.go

```diff
@@ -1731,3 +1732,34 @@ func (h *ChunkServiceHandler) EndSync(ctx context.Context, req *models.EndSyncRe
 
 	return &models.EndSyncResponse{}, nil
 }
+
+func (h *ChunkServiceHandler) GetMVSize(ctx context.Context, req *models.GetMVSizeRequest) (*models.GetMVSizeResponse, error) {
+	// Thrift should not be calling us with nil req.
+	common.Assert(req != nil)
+
+	log.Debug("ChunkServiceHandler::GetMVSize: Received GetMVSize request: %v", rpc.GetMVSizeRequestToString(req))
+
+	if !cm.IsValidMVName(req.MV) || !cm.IsValidRVName(req.RVName) {
+		errStr := fmt.Sprintf("MV (%s) or RV (%s) invalid", req.MV, req.RVName)
+		log.Err("ChunkServiceHandler::GetMVSize: %s", errStr)
+		common.Assert(false, errStr)
+		return nil, rpc.NewResponseError(models.ErrorCode_InvalidRequest, errStr)
+	}
+
+	rvInfo := h.getRVInfoFromRVName(req.RVName)
+	if rvInfo == nil {
+		log.Err("ChunkServiceHandler::GetMVSize: Invalid RV %s", req.RVName)
+		return nil, rpc.NewResponseError(models.ErrorCode_InvalidRV, fmt.Sprintf("invalid RV %s", req.RVName))
+	}
+
+	mvInfo := rvInfo.getMVInfo(req.MV)
+	if mvInfo == nil {
+		errStr := fmt.Sprintf("%s/%s not hosted by this node", rvInfo.rvName, req.MV)
+		log.Err("ChunkServiceHandler::GetMVSize: %s", errStr)
+		return nil, rpc.NewResponseError(models.ErrorCode_NeedToRefreshClusterMap, errStr)
+	}
+
```

Comment:
pls make this consistent with a similar handler, say JoinMV()
make logs, use of errStr, comments, everything consistent.
f.e. we need a comment explaining why it's ok not to retry before returning ErrorCode_NeedToRefreshClusterMap